BDArmory/GameModes/Waypoints/CourseBuilderGUI.cs

```diff
@@ -0,0 +1,679 @@
+                        else
+                            spawnFields["increment"].SetCurrentValue(spawnFields["increment"].currentValue - 0.01);
+                    }
+                    if (spawnFields["increment"].currentValue < 0.001f) spawnFields["increment"].SetCurrentValue(0.001f);
+                }
+                spawnFields["increment"].tryParseValue(GUI.TextField(rects[moddingSpawnPoint ? 0 : 2], spawnFields["increment"].possibleValue, 8, spawnFields["increment"].style));
+                if (GUI.Button(SFieldButtonRect(line, moddingSpawnPoint ? 7 : 22), ">", BDArmorySetup.BDGuiSkin.button))
+                {
+                    spawnFields["increment"].SetCurrentValue(spawnFields["increment"].currentValue + 1);
+                    if (spawnFields["increment"].currentValue > 1000) spawnFields["increment"].SetCurrentValue(1000);
+                }
+                movementIncrement = spawnFields["increment"].currentValue;
+                if (!moddingSpawnPoint)
+                {
+                    if (spawnFields["lat"].currentValue != currGate.location.x ||
+                    spawnFields["lon"].currentValue != currGate.location.y ||
+                    spawnFields["alt"].currentValue != currGate.location.z ||
+                    spawnFields["speed"].currentValue != currGate.maxSpeed ||
+                    spawnFields["diameter"].currentValue != currGate.scale)
+                    {
+                        currGate.location = new Vector3d(spawnFields["lat"].currentValue, spawnFields["lon"].currentValue, spawnFields["alt"].currentValue);
+                        currGate.scale = (float)spawnFields["diameter"].currentValue;
+                        currGate.maxSpeed = (float)spawnFields["speed"].currentValue;
+                        WaypointField.Save(); //instead have a separate button and do saving manually?
+                        loadedGates[selected_gate_index].UpdateWaypoint(currGate, selected_gate_index, WaypointCourses.CourseLocations[selected_index].waypoints);
+                        SnapCameraToGate();
+                    }
+                }
+                else
+                {
+                    if (spawnFields["lat"].currentValue != WaypointCourses.CourseLocations[selected_index].spawnPoint.x ||
+spawnFields["lon"].currentValue != WaypointCourses.CourseLocations[selected_index].spawnPoint.y ||
+spawnFields["alt"].currentValue != BDArmorySettings.VESSEL_SPAWN_ALTITUDE)
+                    {
+                        WaypointCourses.CourseLocations[selected_index].spawnPoint = new Vector2d(spawnFields["lat"].currentValue, spawnFields["lon"].currentValue);
+                        BDArmorySettings.VESSEL_SPAWN_ALTITUDE = (float)spawnFields["alt"].currentValue;
+
+                        WaypointField.Save(); //instead have a separate button and do saving manually?
```

Comment:
`Waypoint_locations.cfg` is saved automatically when `WaypointCourses.Instance` gets destroyed when leaving flight mode and is only loaded when entering flight mode, so there's no real reason to save here.

If you want to save/reload while still within flight mode, then this would be better with separate buttons.